apps/passport-server/src/services/userService.ts

```diff
@@ -56,6 +57,18 @@ export class UserService {
     return user.salt;
   }
 
+  /**
+   * Returns the encryption key for a given user, if it is stored on
+   * our server. Returns null if the user does not exist, or if
+   * the user does not have an encryption key stored on the server.
+   */
+  public async getEncryptionKeyForUser(
+    email: string
+  ): Promise<HexString | null> {
+    const existingUser = await fetchUserByEmail(this.context.dbPool, email);
+    return existingUser?.encryption_key ?? null;
+  }
+
```

Comment:
this feels slightly dangerous. if anyone can request anyone's encryption key, i think anyone can derive anyone's blob key, and download their e2ee storage, right?

Reply:
Definitely dangerous in the absence of authentication.  In this PR do we only ever return this field as part of the login process after the user submits a verification token?  Or is it also returned by the general get_user functionality?  Limiting it to the former is probably safest.  There's no reason a user should need to fetch their key again after login.

Reply:
this is only called in the /verify-token endpoint, after the token is verified. so in order to get a user's encryption_key, they would need to have access to their email account